### ISSUE TITLE
CORS: preflight doesn't return any (*) for access-control-allow-origin using allow_origin(Any)

### ISSUE THREAD

**Issue by lichr** [tower-rs/tower-http#244]
## Bug Report

Preflight doesn't return any (*) for `access-control-allow-origin` when setting `allow_origin(Any)`

### Version

0.2.5

### Platform

Linux snow 5.14.15-300.fc35.x86_64 #1 SMP Wed Oct 27 15:53:39 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

### Description

I tried this code with an axum api service:

```rust
    use tower_http::cors::{Any, CorsLayer};
...
    let cors = CorsLayer::new().allow_methods(Any).allow_origin(Any);
    let app = make_router(shared_state).layer(cors);
...
```
Call it with preflight request:
```http
OPTIONS http://localhost:3000/some-endpoint
Origin: http://localhost:9002
Access-Control-Request-Method: GET
Access-Control-Request-Headers: authorization

```

I expected to see this happen: 
```http
HTTP/1.1 200 OK
access-control-allow-origin: *
access-control-allow-methods: *
```

Instead, this happened:
```http
HTTP/1.1 200 OK
access-control-allow-origin: http://localhost:9002
access-control-allow-methods: *
```

In my case this issue prevents my SPA to access my API (running on another server), even with the SPA's origin returned like shown as above, chrome still raises a CORS error, `access-control-allow-origin: *` would be fine, that is why I would like `.allow_origin(Any)` to be working as intended.

Thanks a lot

**Comment by jplatte** [comment]
This will be fixed by #237, however I don't think that the CORS error you are seeing is related to the allow-origin response header. If the browser is sending access-control-request-headers: authorization, you need to respond with access-control-allow-headers: authorization for the request to be allowed. A wildcard for headers excludes authorization IIRC.

**Comment by lichr** [comment]
@jplatte You are right, the CORS error goes away after adding access-control-allow-headers: * to preflight response, thanks you for pointing this out !

**Comment by bnjjj** [comment]
Hello, #237 looks great ! I'm also waiting for this PR to solve the issue https://github.com/apollographql/router/issues/792

**Comment by jplatte** [comment]
Fixed by #237.
